### PR TITLE
Baroclinic eddies test

### PR DESCRIPTION
**Pull request by tkarna**
Updates for baroclinic eddies test.
Also added new eddying channel test by Danilov (2012).

### CODE REVIEW

**Comment by tkarna** [comment]
I think this is ready to go now.

File: examples/baroclinic_channel/baroclinic_channel.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+physical_constants['rho0'] = 1020.
+
+refinement = 1  # normal = 4
+lx = ly = 1600e3
+nx = ny = 43*refinement
+delta_x = lx/nx
+delta_y = ly/ny
+mesh2d = PeriodicRectangleMesh(nx, ny, lx, ly, direction='x')
+depth = 1600.
+nlayers = 10  # FIXME
```
Comment (stephankramer) [inline examples/baroclinic_channel/baroclinic_channel.py]: Does this need fixing indeed?

Comment (tkarna) [reply examples/baroclinic_channel/baroclinic_channel.py]: Yes, Danilov uses 24 levels, I used 10 for testing. I'll revise this upstream.

**Comment by tkarna** [comment]
Yes, this is the baroclinic eddies test used in the Thetis paper.

**Pull request merged**